packages/contracts-bedrock/contracts/L1/ResourceMetering.sol

```diff
@@ -104,20 +107,16 @@ abstract contract ResourceMetering is Initializable {
                 // Update the base fee by repeatedly applying the exponent 1-(1/change_denominator)
                 // blockDiff - 1 times. Simulates multiple empty blocks. Clamp the resulting value
                 // between min and max.
-                newBaseFee = SignedMath.min(
-                    SignedMath.max(
-                        int256(
-                            (newBaseFee *
-                                (
-                                    FixedPointMathLib.powWad(
-                                        1e18 - (1e18 / BASE_FEE_MAX_CHANGE_DENOMINATOR),
-                                        int256((blockDiff - 1) * 1e18)
-                                    )
-                                )) / 1e18
-                        ),
-                        int256(MINIMUM_BASE_FEE)
-                    ),
-                    int256(uint256(type(uint128).max))
+                newBaseFee = Arithmetic.clamp(
+                    (newBaseFee *
```

Comment:
Can we turn line 111 into a single line by putting most of the logic into a function?

Reply:
if we make 111 into a single line by making a lib fn that wraps the ugly nesting then 100% will approve this

Reply:
Done

Reply:
clean!